lib/nxt_pipeline/dsl.rb

```diff
@@ -0,0 +1,44 @@
+module NxtPipeline
+  module Dsl
+    module ClassMethods
+      def pipeline(name = :default, &block)
+        name = name.to_sym
+
+        if block_given?
+          pipeline_registry.key?(name) && raise_already_registered_error(name)
+          pipeline_registry[name] = block
+        else
+          config = pipeline_registry.fetch(name) { raise KeyError, "No pipeline #{name} registered"}
+          NxtPipeline::Pipeline.new(&config)
+        end
+      end
+
+      def pipeline!(name, &block)
+        raise ArgumentError, "No block given!" unless block_given?
+        pipeline_registry[name] = block
+      end
+
+      private
+
+      def inherited(child)
+        child.instance_variable_set(:@pipeline_registry, pipeline_registry.clone)
```

Comment:
Just to ensure I understand it correctly: This ensure that you can register pipelines in a parent class and inherit this registry content in a subclass while being able to append new pipelines into the registry in the subclass without also putting them into the parent's registry, right?

Reply:
Yes exactly. You will still be able to overwrite with ´pipeline!´ if you would want to reconfigure the inherited pipeline steps.

Reply:
Nice! 👍